centreon/src/Core/HostCategory/Application/UseCase/CreateHostCategory/CreateHostCategory.php

```diff
@@ -0,0 +1,106 @@
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+* For more information : contact@centreon.com
+*
+*/
+
+declare(strict_types=1);
+
+namespace Core\HostCategory\Application\UseCase\CreateHostCategory;
+
+use Centreon\Domain\Contact\Contact;
+use Centreon\Domain\Contact\Interfaces\ContactInterface;
+use Centreon\Domain\Log\LoggerTrait;
+use Core\Application\Common\UseCase\ErrorResponse;
+use Core\Application\Common\UseCase\ForbiddenResponse;
+use Core\Application\Common\UseCase\InvalidArgumentResponse;
+use Core\Application\Common\UseCase\PresenterInterface;
+use Core\HostCategory\Application\Exception\HostCategoryException;
+use Core\HostCategory\Application\Repository\ReadHostCategoryRepositoryInterface;
+use Core\HostCategory\Application\Repository\WriteHostCategoryRepositoryInterface;
+use Core\HostCategory\Application\UseCase\CreateHostCategory\CreateHostCategoryRequest;
+use Core\HostCategory\Domain\Model\HostCategory;
+use Core\HostCategory\Domain\Model\NewHostCategory;
+
+final class CreateHostCategory
+{
+    use LoggerTrait;
+
+    public function __construct(
+        private WriteHostCategoryRepositoryInterface $writeHostCategoryRepository,
+        private ReadHostCategoryRepositoryInterface $readHostCategoryRepository,
+        private ContactInterface $user
+    ) {
+    }
+
+    public function __invoke(CreateHostCategoryRequest $request, PresenterInterface $presenter): void
+    {
+        try {
+            if (! $this->user->hasTopologyRole(Contact::ROLE_CONFIGURATION_HOSTS_CATEGORIES_READ_WRITE)) {
+                $this->error('User doesn\'t have sufficient right to see host categories', [
+                    'user_id' => $this->user->getId(),
+                ]);
+                $presenter->setResponseStatus(
+                    new ForbiddenResponse(HostCategoryException::createNotAllowed()->getMessage())
+                );
+            } elseif ($this->readHostCategoryRepository->existsByName(trim($request->name))) {
+                $this->error('Host category name already exists', [
+                    'hostcategory_name' => trim($request->name),
+                ]);
+                $presenter->setResponseStatus(
+                    new InvalidArgumentResponse(HostCategoryException::hostNameAlreadyExists()->getMessage())
+                );
+            } else {
+                $newHostCategory = new NewHostCategory(trim($request->name), trim($request->alias));
+                $newHostCategory->setComment($request->comment ? trim($request->comment) : null);
+
+                $hostCategoryId = $this->writeHostCategoryRepository->add($newHostCategory);
+                $hostCategory = $this->readHostCategoryRepository->findById($hostCategoryId);
+
+                $presenter->present(
+                    // new CreatedResponse($hostCategoryId, $this->createResponse($hostCategory))
+                    $this->createResponse($hostCategory ?? null)
```

Comment:
check the single [commit e69982a2](https://github.com/centreon/centreon/commit/e69982a2f73630dd074892526ae16e6e1c460f06) I made for that from Laurent's PR #315 if it can help you.

Reply:
WIP